docs/source/developers/guide/tutorials/r_tutorial.rst

```diff
@@ -22,21 +22,429 @@
+is already specified/mapped on the R package side:
+`<https://github.com/apache/arrow/blob/658bec37aa5cbdd53b5e4cdc81b8ba3962e67f11/r/R/expression.R#L63-L64>`_
+
+We only need to add ``mday`` to the list of expressions connecting
+it to the C++ ``day`` function.
+
+.. code-block:: R
+
+   # second is defined in dplyr-functions.R
+   # wday is defined in dplyr-functions.R
+   "mday" = "day",
+   "yday" = "day_of_year",
+   "year" = "year",
+
+Adding a test
+-------------
+
+Now we need to add a test that checks if everything works well.
+If there are additional options or edge cases, we would have to
+add more. Looking at tests for similar functions (for example
+``yday()`` or ``day())`` we can see that a good place to add two
+tests we have is in ``test-dplyr-funcs-datetime.R``:
+
+.. code-block:: R
+
+   test_that("extract mday from timestamp", {
+     compare_dplyr_binding(
+       .input %>%
+         mutate(x = mday(datetime)) %>%
+         collect(),
+       test_df
+     )
+   })
+
+And 
+
+.. code-block:: R
+
+   test_that("extract mday from date", {
+     compare_dplyr_binding(
+       .input %>%
+         mutate(x = mday(date)) %>%
+         collect(),
+       test_df
+     )
+   })
+
+Now we need to see if the tests are passing or we need to do some
+more research and code corrections.
+
+.. code-block:: R
+
+   devtools::test(filter="datetime")
+
+   > devtools::test(filter="datetime")
+   ℹ Loading arrow
+   See arrow_info() for available features
+   ℹ Testing arrow
+   See arrow_info() for available features
+   ✔ | F W S  OK | Context
+   ✖ | 1     230 | dplyr-funcs-datetime [1.4s]                                                                                             
+   ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
+   Failure (test-dplyr-funcs-datetime.R:187:3): strftime
+   ``%>%`(...)` did not throw the expected error.
+   Backtrace:
+    1. testthat::expect_error(...) test-dplyr-funcs-datetime.R:187:2
+    2. testthat:::expect_condition_matching(...)
+   ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
+
+   ══ Results ═════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════
+   Duration: 1.4 s
+
+   [ FAIL 1 | WARN 0 | SKIP 0 | PASS 230 ]
+
+There is a fail we get for the ``strftime`` function but looking
+at the code we see is not connected to our work. We can move on and
+maybe ask others if they are getting similar fail when running the tests.
+It could be we only need to rebuild the library.
+
+Check styling
+-------------
+
+As we did not change any part of the C++ code we will not
+run any linters. In case we would need to check the styling
+for the C++ part we would run ``arrow/r/lint.sh`` to lint the code.
```

Comment:
We still need to run `make style` in the `r` directory to style the R code

Reply:
Will add this, thanks for pointing it out!